services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ControllerRoutes.java

```diff
@@ -179,6 +190,44 @@ private Route updateKafkaTopicConfig(Admin admin, UpdateTopicConfigFunction upda
     };
   }
 
+  public Route getAggregatedHealthStatus(Admin admin) {
+    return (request, response) -> {
+      StoppableNodeStatusResponse responseObject = new StoppableNodeStatusResponse();
+      response.type(HttpConstants.JSON);
+
+      try {
+        String cluster = request.queryParams(CLUSTER_ID);
+        String instances = request.queryParams(INSTANCES);
+        String toBeStoppedInstances = request.queryParams(TO_BE_STOPPED_INSTANCES);
```

Comment:
It should, check other post APIs.

Reply:
Can you verify this through curl or other HTTP request tooling while integration tests are running? Our implementation seems to be using `x-www-urlencoded` that sparkjava will parse as query params. I didn't find the handling for JSON inputs in Spark java documentation. So, it doesn't look like it'll work by default

Reply:
It does come in as query parameters. Our implementation of executing a post request seems a bit strange with putting the `QueryParams` as part of the url and setting the `byte[]` payload if any as part of the post request body.

I'd suggest give the instances and `not_being_stopped` takes a set that can have multiple items (order of 10s -> potentially 100) as our cluster grows, it would be wise to use this the post payload to pass this information.

Check out other APIs that does this; `sendPushJobDetails` is an example on how this done; You will need to have a model and a serializer ahead and pass the serialized bytes as part of the `request` method